src/departure/departureBlock/departureBlocksOMMDataSource.ts

```diff
@@ -0,0 +1,392 @@
+  let sslAgent = new https.Agent({
+    host: OMM_API_HOST,
+    port: parseInt(OMM_API_PORT, 10),
+    path: OMM_API_PATH,
+    rejectUnauthorized: false, // Don't mind self-signed certs.
+  })
+
+  let ommOperatorId = (operatorId + '').padStart(4, '0')
+
+  let ommResponse = await fetch(
+    `https://${OMM_API_HOST}:${OMM_API_PORT}/${OMM_API_PATH}?operator=${ommOperatorId}&from=${startDate}&to=${endDate}&hastusData=${
+      hastusData ? 'true' : 'false'
+    }`,
+    {
+      method: 'GET',
+      agent: sslAgent,
+      headers: {
+        username: OMM_API_USERNAME,
+        organisation: OMM_API_ORGANISATION,
+        Authorization: OMM_API_BASIC_AUTH,
+      },
+    }
+  )
+
+  if (ommResponse.status >= 400) {
+    throw new ApolloError(`Error from OMM block service: ${ommResponse.statusText}`)
+  }
+
+  let ommResponseBody = await ommResponse.text()
+  return JSON.parse(ommResponseBody) || []
+}
+
+/**
+ * Fetch departure blocks from OMM (or test data for test operator) and return
+ * OperatorBlockDeparture entities based on them.
+ * @param inspection
+ * @param useCache
+ */
+export async function getDepartureBlocksForInspection(
+  inspection: PreInspection,
+  useCache: boolean = true
+): Promise<OperatorBlockDeparture[]> {
+  let { id, inspectionStartDate, inspectionEndDate, operatorId } = inspection
+  let cacheKey = `${id}_${inspectionStartDate}_${inspectionEndDate}_${operatorId}`
+
+  let blockDepartureEntities = await cacheFetch<OperatorBlockDeparture[]>(
+    cacheKey,
+    async () => {
+      let response = await fetchDepartureBlocks({
+        startDate: inspectionStartDate,
+        endDate: inspectionEndDate,
+        operatorId,
+      })
+
+      if (!response || !Array.isArray(response) || response.length === 0) {
+        return []
+      }
+
+      return parseBlockDeparturesFromOMMResponse(response, inspection)
+    },
+    {
+      ttl: 60 * 60,
+      forceFetch: !useCache,
+    }
+  )
+
+  return blockDepartureEntities || []
+}
+
+/**
+ * Takes OMM block items, parses the hastusData and returns OperatorBlockDeparture entities.
+ * @param ommBlockItems
+ * @param inspection
+ */
+async function parseBlockDeparturesFromOMMResponse(
+  ommBlockItems: OMMBlockItem[] = [],
+  inspection: PreInspection
+): Promise<OperatorBlockDeparture[]> {
+  let inspectionDates = eachDayOfInterval({
+    start: getDateObject(inspection.inspectionStartDate),
+    end: getDateObject(inspection.inspectionEndDate),
+  })
+
+  let blockDepartures: OperatorBlockDeparture[] = []
+
+  // Find the OMM block item valid for each date and use its hastusData to create OperatorBlockDeparture entities.
+  for (let inspectionDate of inspectionDates) {
+    let dateStr = format(inspectionDate, 'dd.MM.yyyy')
+
+    // Find OMM items valid for this date.
+    let ommItemsForDate = ommBlockItems.filter(
+      ommItem =>
+        !ommItem?.exceptionDate && ommItem.validOnDates.some(validDate => validDate === dateStr)
```

Comment:
```suggestion
        ommItem.validOnDates.some(validDate => validDate === dateStr)
```

I think we should let departure blocks through that are both exceptionDate=false and exceptionDate=true. Isn't it user's responsibility to choose a week that has no exception dates? (This will be done in part 2 of this task)

Reply:
The pre-inspection does not currently take exception dates into account for wither block or JORE departures. When/if we change it so that it does, this filter should be removed. This should probably be done in part 2, but it is not that important as long as it is OK that exception dates are not taken into account.